demo/src/app/components/rating/demos/decimal/decimal.component.html

```diff
@@ -0,0 +1,12 @@
+<template #t let-fill="fill">
+  <span *ngIf="fill === 100" class="star full">&hearts;</span>
+  <span *ngIf="fill === 0" class="star">&hearts;</span>
+  <span *ngIf="fill < 100 && fill > 0" class="star">
+    <span class="half" style.width="{{fill + '%'}}">&hearts;</span>&hearts;
```

Comment:
Kinda works both ways `[style.width]=""` and `style.width='{{}}'`, but will try the one @pkozlowski-opensource is suggesting!